pretyping/typeclasses.mli

```diff
@@ -13,7 +13,7 @@ open Constr
 open Evd
 open Environ
 
-type direction = Forward | Backward
+type direction = Backward
```

Comment:
What's this used for?

Reply:
Nothing, I could remove it but then it's more invasive API-wise.

Reply:
Invade on imo

Reply:
Done. I took the occasion to introduce a record type for methods.